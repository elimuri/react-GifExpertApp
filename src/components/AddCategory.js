import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        console.log("HandleInputChange lamado")
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit llamado', inputValue)

        if(inputValue.trim()){
            setCategories((categories) => [inputValue, ...categories]);
            setInputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
          <p>{inputValue}</p>
          <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}