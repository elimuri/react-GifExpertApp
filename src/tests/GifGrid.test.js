import { shallow } from "enzyme"
import React from 'react'
import { GifGrid } from "../components/GifGrid";

describe('Pruebas en <AddCategory />', () => {
    let wrapper = shallow(<GifGrid category={'Hollywood monsters'} />);

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
});