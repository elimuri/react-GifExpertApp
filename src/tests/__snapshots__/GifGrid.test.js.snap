// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Hollywood monsters
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
